iron-fit-behavior.html

```diff
@@ -349,37 +355,21 @@
       var positionRect = this.__getNormalizedRect(this.positionTarget);
       var fitRect = this.__getNormalizedRect(this.fitInto);
 
-      var margin = this._fitInfo.margin;
+      var position = this.__getPosition(this._localeHorizontalAlign, this.verticalAlign, rect, positionRect, fitRect);
 
-      // Consider the margin as part of the size for position calculations.
-      var size = {
-        width: rect.width + margin.left + margin.right,
-        height: rect.height + margin.top + margin.bottom
-      };
-
-      var position = this.__getPosition(this._localeHorizontalAlign, this.verticalAlign, size, positionRect, fitRect);
+      var margin = this._fitInfo.margin;
+      var minLeft = margin.left + (position.horizontalAlign === 'left' ? this.horizontalOffset : 0);
+      var minTop = margin.top + (position.verticalAlign === 'top' ? this.verticalOffset : 0);
+      var maxRight = fitRect.right - margin.right - (position.horizontalAlign === 'left' ? 0 : this.horizontalOffset);
+      var maxBottom = fitRect.bottom - margin.bottom - (position.verticalAlign === 'top' ? 0 : this.verticalOffset);
 
       var left = position.left + margin.left;
       var top = position.top + margin.top;
+      var right = Math.min(maxRight, left + rect.width);
+      var bottom = Math.min(maxBottom, top + rect.height);
 
-      // Use original size (without margin).
-      var right = Math.min(fitRect.right - margin.right, left + rect.width);
-      var bottom = Math.min(fitRect.bottom - margin.bottom, top + rect.height);
-
-      var minWidth = this._fitInfo.sizedBy.minWidth;
-      var minHeight = this._fitInfo.sizedBy.minHeight;
-      if (left < margin.left) {
-        left = margin.left;
-        if (right - left < minWidth) {
-          left = right - minWidth;
-        }
-      }
-      if (top < margin.top) {
-        top = margin.top;
-        if (bottom - top < minHeight) {
-          top = bottom - minHeight;
-        }
-      }
+      left = Math.min(Math.max(left, minLeft), right - this._fitInfo.sizedBy.minWidth);
+      top = Math.min(Math.max(top, minTop), bottom - this._fitInfo.sizedBy.minHeight);
```

Comment:
Uhm, not really; we need the first value of `left` (line 366) to compute `right` (line 368, already limited to not pass `maxRight`)
top-left example:
<img width="558" alt="screen shot 2016-07-18 at 3 41 10 pm" src="https://cloud.githubusercontent.com/assets/6173664/16932850/14502b38-4cfe-11e6-9859-9c7a7d3de860.png">

top-right example:
<img width="542" alt="screen shot 2016-07-18 at 3 47 52 pm" src="https://cloud.githubusercontent.com/assets/6173664/16932992/143a8034-4cff-11e6-87f7-0966f64972c0.png">

Only then we can limit the `left` to not pass `minLeft` and to respect `minWidth` (it will pass `minLeft` if there is a `minWidth > 0`). This allows to have an item aligned to the left with a horizontal scrollbar.

Top-left example:
<img width="559" alt="screen shot 2016-07-18 at 3 41 27 pm" src="https://cloud.githubusercontent.com/assets/6173664/16932854/226592c6-4cfe-11e6-806c-2f48d0d48acf.png">

Top-right examples:
<img width="545" alt="screen shot 2016-07-18 at 3 49 11 pm" src="https://cloud.githubusercontent.com/assets/6173664/16933016/3b1c2928-4cff-11e6-9380-46f137f07bfc.png">
<img width="549" alt="screen shot 2016-07-18 at 3 49 19 pm" src="https://cloud.githubusercontent.com/assets/6173664/16933025/406e04a0-4cff-11e6-9c51-99426e24ddf9.png">


Moving the `max` to line 366 would always keep the element aligned to the left and never show the horizontal scrollbar.